unity-client/Assets/Shaders/MaterialTransition/Unlit Hologram.shader

```diff
@@ -77,6 +79,19 @@ Shader "DCL/FX/Hologram"
 
             float4 frag(vertexOutput i) : COLOR
             {
+                float distanceAlphaMultiplier = 1;
+
+                if(_MaxRenderingDistance > 0)
+                {
+                    float renderingDistance = length(_WorldSpaceCameraPos.xyz - i.posWorld.xyz);
+
+                    if (renderingDistance > _MaxRenderingDistance) return (0,0,0,0);
+
+                    float halfMaxRenderingDistance = _MaxRenderingDistance / 2;
+                    if (renderingDistance > halfMaxRenderingDistance)
+                        distanceAlphaMultiplier = 1 - ((renderingDistance - halfMaxRenderingDistance) / halfMaxRenderingDistance);
+                }
```

Comment:
Why are we using the `_MaxRenderingDistance` if we just halve it later? I think its a bit better to let the frustum culling do the far away culling, and just do the `distanceAlphaMultiplier` bit with a `saturate` to clamp it to 0--1. This way we can save a couple of branching instructions that are bad for shader performance.

Reply:
The current logic does the follwing:
* has a max rendering distance for the hologram
* In the farer half-distance, the alpha multiplier is lerped/applied
* In the nearest half-distance the alpha multiplier is always 1

`i.pos.z` didn't work out-of-the-box for me but I can try again, and also I would prefer to keep this separated from the frustrum culling in case we some day change the frustrum culling, but if you insist I can try changing that to use what you mentioned

Reply:
I just meant that we can save the branching that's a bit slow on older GPUs. The distance logic is fine!

I suggest that we remove all the if statements with the exception of the early exit one:
```
                    float renderingDistance = length(_WorldSpaceCameraPos.xyz - i.posWorld.xyz);

                    if (renderingDistance > _MaxRenderingDistance) return (0,0,0,0);

                    float halfMaxRenderingDistance = _MaxRenderingDistance / 2;
                    distanceAlphaMultiplier = 1 - ((renderingDistance - halfMaxRenderingDistance) / halfMaxRenderingDistance);
```

This said, having the `halfMaxRenderingDistance` as the fade distance threshold irks me a bit, but its fine for our purposes.

Reply:
Oh I understand now, great, yes, I'll make that change

Reply:
Done